type: fix EKS Katib failing test

deusebio opened the pull request:
Fixes #1210 

Investigation of the issue showed that trial pods for Katib consistently fail on EKS because of `metrics-logger-and-collector` container erroring out with the following message

```
(base) jovyan@workload-test-0:~$ kubectl logs -n 200 cmaes-example-4rm4bmt2-mkhrj -n admin -c metrics-logger-and-collector
I0224 12:38:04.158021      28 main.go:396] Trial Name: cmaes-example-4rm4bmt2
100.0%12:38:04.158578      28 main.go:139] 
100.6%12:38:04.160186      28 main.go:139] 
100.0%12:38:04.160318      28 main.go:139] 
119.3%12:38:04.160379      28 main.go:139] 
2025/02/24 12:38:04 FATAL -- failed to create Watcher
goroutine 6 [running]:
runtime/debug.Stack()
    /usr/local/go/src/runtime/debug/stack.go:24 +0x5e
github.com/hpcloud/tail/util.Fatal({0xdce7f6?, 0x411a3b?}, {0x0, 0x0, 0x0})
    /go/pkg/mod/github.com/hpcloud/tail@v1.0.1-0.20180514194441-a1dbeea552b7/util/util.go:22 +0x8b
github.com/hpcloud/tail/watch.(*InotifyTracker).run(0xc00004a140)
    /go/pkg/mod/github.com/hpcloud/tail@v1.0.1-0.20180514194441-a1dbeea552b7/watch/inotify_tracker.go:219 +0x65
created by github.com/hpcloud/tail/watch.init.func1 in goroutine 5
    /go/pkg/mod/github.com/hpcloud/tail@v1.0.1-0.20180514194441-a1dbeea552b7/watch/inotify_tracker.go:54 +0x14e
```

As mentioned [here](https://github.com/kubeflow/katib/issues/2434), this may be due to the `too many open files` exception. 

Indeed setting the sysctl properties 

```
sudo sysctl fs.inotify.max_user_instances=1280
sudo sysctl fs.inotify.max_user_watches=655360
```

on the EKS worker nodes seems to fix the issue (see [here](https://github.com/canonical/bundle-kubeflow/actions/runs/13504705058))

I have also checked that AKS does not seem to be affected by the same issue, see [here](https://github.com/canonical/bundle-kubeflow/actions/runs/13518298872/job/37771632988)

orfeas-k commented on .github/workflows/deploy-to-eks.yaml:
```diff
@@ -102,9 +102,48 @@ jobs:
+            $SSH_CMD 'sudo sysctl fs.inotify.max_user_watches=655360'
+            $SSH_CMD 'sudo sysctl fs.inotify.max_user_instances=1280'
+          
+            # To check the settings
+            echo "Checking values"
+            $SSH_CMD 'sudo sysctl -a | grep "^fs.inotify"'
+            echo "=========================================="
```
(this is a general comment but adding it here to facilitate the discussion) Should we implement also a mechanism to catch when the script has failed to set the values properly? An example for that would be a change in the output of the `aws ec2 describe-instances` command causes the dns query to not extract the required information. We 'd then use this mechanism to throw an error in the CI, so this doesn't go unnoticed

orfeas-k commented on .github/workflows/deploy-to-eks.yaml:
```diff
@@ -102,9 +102,48 @@ jobs:
           kubectl get nodes
-      
+
+      - name: Configure EKS Nodes
+        run: |
+          
```
**suggestion:** Should we put all of this in a script probably under https://github.com/canonical/bundle-kubeflow/tree/main/scripts/gh-actions? This way, it would be more reusable (ie run the script locally if needed)

deusebio replied:
yes, that's a good idea. done

orfeas-k reviewed (commented):
Good job @deusebio , specially on the thorough investigation that this required. The existing implementation looks good, I 've left some general comments. On a final note, this makes me think if we should add a note about this requirement in the UATs, to ensure people have their VMs configured with the proper setting. It's a bit awkward because it depends on the specific k8s substrate, but in k8s that use `inotify`, this is a CKF requirement. Currently, the only place it is mentioned is in the [get started guide](https://charmed-kubeflow.io/docs/get-started).

deusebio commented:
I have addressed the comments. On the following comment

> On a final note, this makes me think if we should add a note about this requirement in the UATs,

I believe that's what we are already doing with having the katib test (although it may be made more robust that when things fail, we actually spot that). Implementing a check in the UAT is a bit more complex, since you would like to make it non substrate dependent. Already here, in order to this to work, the EKS cluster needs to be deployed with certain features, e.g. ssh enabled

deusebio commented:
Tested the new modification which are working, see [here](https://github.com/canonical/bundle-kubeflow/actions/runs/13545529561/job/37856124508#step:11:33)

orfeas-k commented:
> I believe that's what we are already doing with having the katib test (although it may be made more robust that when things fail, we actually spot that). Implementing a check in the UAT is a bit more complex, since you would like to make it non substrate dependent. 

+1, my suggestion was more like if we have to document this somewhere more clearly as a requirement, not that the UATs should be responsible for this check, it's an infrastructure check in the end while UATs check the components' functionality.

> although it may be made more robust that when things fail, we actually spot that

Should we have an issue in the UATs repo for that actually? Having false positive checks sounds quite important

orfeas-k reviewed (approved):
Thank you @deusebio for the neat work, approving! Let's add a `type:` in the merged commit message.

mvlassis commented on scripts/gh-actions/set_eks_sysctl_config.sh:
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
```
We could add `set -xe` to immediately exit on an error in the script
```suggestion
set -xe
```
